hazelcast/src/test/java/com/hazelcast/sql/impl/type/converter/ConvertersTest.java

```diff
@@ -263,13 +278,18 @@ public void testBigDecimalConverter() {
         checkConverterConversions(converter, VARCHAR, TINYINT, SMALLINT, INT, BIGINT, REAL, DOUBLE, OBJECT);
 
         BigDecimal val = BigDecimal.valueOf(11, 1);
+        BigDecimal bigValue = BigDecimal.valueOf(Long.MAX_VALUE).add(new BigDecimal("1.1"));
 
         assertEquals("1.1", converter.asVarchar(val));
 
         assertEquals(1, converter.asTinyint(val));
+        checkDataException(() -> converter.asTinyint(bigValue));
```

Comment:
Might be useful to test a conversion with precision loss: 
```java
assertEquals(1, converter.asTinyint(BigDecimal.valueOf(1.1)));
```

Reply:
Looks like it's already tested at https://github.com/hazelcast/hazelcast/pull/17183/files/1f0b562628e25e8cd2941ff956afedd5fd8c5d42#diff-547f6c79d281b453fd7d6988b2bb7096R285, `val = BigDecimal.valueOf(11, 1) = 1.1`